jobbergate-agent/jobbergate_agent/internals/update.py

```diff
@@ -0,0 +1,71 @@
+import subprocess
+import sys
+
+from loguru import logger
+from pkg_resources import get_distribution
+
+from jobbergate_agent.clients.cluster_api import backend_client as jobbergate_api_client
+from jobbergate_agent.utils.scheduler import schedule_tasks, scheduler
+
+
+package_name = "jobbergate_agent"
+
+
+async def _fetch_upstream_version_info() -> str:
+    logger.debug("Fetching the upstream version info from jobbergate API")
+    response = await jobbergate_api_client.get("/jobbergate/openapi.json")
+    response.raise_for_status()
+
+    data = response.json()
+    upstream_version: str = data["info"]["version"]
+    return upstream_version
+
+
+def _compare_versions(current_version: str, upstream_version: str) -> bool:
+    """Compare the current version with the upstream version.
+
+    In case the current version is the same as the upstream version, return False.
+    As well as, in case the major versions are the same, return False, as we don't want
+    to update across major versions. Otherwise, return True.
+
+    This behaviour allows the agent to update and rollback across all minor and patch versions.
+    """
+    if current_version == upstream_version:
+        return False
+    current_major, _, _ = map(int, current_version.split("."))
+    upstream_major, _, _ = map(int, upstream_version.split("."))
+    if current_major != upstream_major:
+        return False
+    return True
+
+
+def _update_package(version: str) -> None:
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "--upgrade", f"{package_name}=={version}"])
```

Comment:
Nice touch. I though that selecting the correct environment to update could be tricky, but `sys.executable` solves the issue.